superset/reports/commands/execute.py

```diff
@@ -172,9 +174,14 @@ def _get_screenshot(self) -> ScreenshotData:
             )
         image_url = self._get_url(user_friendly=True)
         user = self._get_screenshot_user()
-        image_data = screenshot.compute_and_cache(
-            user=user, cache=thumbnail_cache, force=True,
-        )
+        try:
+            image_data = screenshot.get_screenshot(user=user)
+        except SoftTimeLimitExceeded as ex:
+            raise ReportScheduleScreenshotTimeout()
+        except Exception as ex:
+            raise ReportScheduleScreenshotFailedError(
+                f"Failed taking a screenshot {str(ex)}"
+            )
```

Comment:
I'm stealing codecov's comment here, but could we add a test for this broad exception case, too?